src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt

```diff
@@ -158,18 +158,33 @@ abstract class MethodFingerprint(
             for (index in 0 until instructionLength) {
                 var patternIndex = 0
                 var threshold = fingerprintFuzzyPatternScanThreshold
+                var offset = 0;
 
-                while (index + patternIndex < instructionLength) {
+                scan@ while (index + patternIndex < instructionLength) {
                     val originalOpcode = instructions.elementAt(index + patternIndex).opcode
-                    val patternOpcode = pattern.elementAt(patternIndex)
+                    val patternOpcode = pattern.elementAt(patternIndex + offset)
 
                     if (patternOpcode != null && patternOpcode.ordinal != originalOpcode.ordinal) {
                         // reaching maximum threshold (0) means,
                         // the pattern does not match to the current instructions
                         if (threshold-- == 0) break
+
+                        // look ahead
+                        for (depth in 1 until threshold + 2) {
```

Comment:
Is `threshold + 2` arbitrary?

Reply:
Kind of, I wanted to make it so that at a minimum at least the next instruction is considered. If threshold is 0 at that point, you'll only enter that loop once since you start at 1 and would go 'until' 2 (exclusive).

Reply:
@oSumAtrIX